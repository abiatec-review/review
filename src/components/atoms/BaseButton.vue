<template>
  <button v-bind="$attrs" @click="$emit('fetchNamedSearch')">
    <slot />
  </button>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'BaseButton',
});
</script>

<style scoped lang="scss">
button {
  height: 2.325rem;
  font-size: 1rem;
  background: #767676;
  border-radius: 0.25rem;
  padding: 0 1rem;
  outline: 0;
  border: 0;
  color: #fff;
  transition: background-color 100ms ease-in-out;
  cursor: pointer;
}

button:hover {
  background: #767676c0;
}
</style>
