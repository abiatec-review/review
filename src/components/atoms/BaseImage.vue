<template>
  <img :class="logo ? 'logo' : 'icon'" :src="getImgUrl" :alt="alt" />
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue';

export default defineComponent({
  name: 'BaseImage',
  props: {
    imagePath: String,
    logo: Boolean,
    alt: String,
  },
  setup(props) {
    const getImgUrl = computed(() => {
      if (props.logo) {
        // eslint-disable-next-line
        return require(`@/assets/${props.imagePath}`);
      }
      return props.imagePath;
    });

    return { getImgUrl };
  },
});
</script>

<style scoped lang="scss">
.icon {
  width: 200px;
  border-radius: 40%;
}

.logo {
  position: absolute;
  top: 1.5rem;
  left: 1.5rem;
  width: 5rem;
  height: 5rem;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .logo {
    animation: logo-spin infinite 20s linear;
  }
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
</style>
