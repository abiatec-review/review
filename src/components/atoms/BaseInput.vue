<template>
  <input v-bind="$attrs" v-model="theModel" />
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue';

export default defineComponent({
  name: 'BaseInput',
  emits: ['update:modelValue'],
  props: {
    modelValue: {
      type: String,
    },
  },
  setup(props, { emit }) {
    const theModel = computed({
      get: () => props.modelValue,
      set: (value) => emit('update:modelValue', value),
    });

    return { theModel };
  },
});
</script>

<style scoped lang="scss">
input {
  height: 2.325rem;
  font-size: 1.5rem;
  border-radius: 0.25rem;
  outline: 0;
  border: 0.1rem solid #767676;
  margin-right: 1rem;
}
</style>
