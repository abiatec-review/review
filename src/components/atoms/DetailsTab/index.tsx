import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router';
import { toast } from 'react-toastify';
import { getDetails } from 'redux/selectors/details';
import { resetDetails, fetchDetails } from 'redux/actions/details';
import {Spinner} from 'components/atoms';

interface DetailsTabProps {
  episodes: string[]
}


const DetailsTab = ({ episodes }: DetailsTabProps) => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const { episodesInfo, charactersPictures, isLoaded, loadedWithError } = useSelector(getDetails);


  useEffect(() => {

    dispatch(fetchDetails(episodes));

    return () => {
      dispatch(resetDetails());
    };

  }, []);


  if (loadedWithError) {
    toast.error('Something went wrong!', {
      position: "top-right",
      autoClose: 5000,
      hideProgressBar: false,
      closeOnClick: true,
      pauseOnHover: true,
      draggable: true,
      progress: undefined,
    });
    navigate(-1);
  }

  return (
    <div>
      {!isLoaded ? 
      <Spinner/>
      
      :
        <ul>
          {episodes.map((episode) => (
            <li key={episode} className='flex justify-between'>
              <span className="w-[150px]">
                {episodesInfo[episode].name}
                <br />
                {episodesInfo[episode].date}
              </span>
              <span>
                {episode}
              </span>
              <span className='flex'>
                {episodesInfo[episode].characters.map((picture: any) => (
                  <img className='mx-4 h-[40px]' key={picture} src={charactersPictures[picture]} alt="" width="40px" height="40px" />
                ))}
              </span>
            </li>
          ))}
        </ul>
      }
    </div>
  );
}

export default DetailsTab;
