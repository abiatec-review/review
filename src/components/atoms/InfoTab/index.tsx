import { PictureType } from 'types';

interface IProps {
    info: PictureType
}

const InfoTab: React.FC<IProps> = ({info}) => {
    const { name,
        status,
        species,
        type,
        gender,
        origin,
        location } = info;

    return (
        <div>
            <figure >
                <img
                    className=' m-auto'
                    src={info.image} alt={`${info.name} character`} />
                <figcaption>{info.name}</figcaption>
            </figure>
            <ul>
                <li>
                    {`name : ${name}`}
                </li>
                <li>
                    {`status : ${status}`}
                </li>
                <li>
                    {`species : ${species}`}
                </li>
                {type &&
                    <li>
                        {`type : ${type}`}
                    </li>
                }
                <li>
                    {`gender : ${gender}`}
                </li>
                <li>
                    {`origin : ${origin.name}`}
                </li>
                <li>
                    {`location : ${location.name}`}
                </li>

            </ul>
        </div>
    );
}

export default InfoTab;
