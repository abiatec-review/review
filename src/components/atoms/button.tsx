import React from "react";

import {
  Pressable,
  PressableProps,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  ViewStyle
} from "react-native";

import { Color, FontSize, Indent, Radius } from "@utils";
interface Props {
  text: string;
  textStyle?: TextStyle;
}

type ButtonProps = Omit<PressableProps, "children"> & Props;

export function Button(props: ButtonProps) {
  const { text, textStyle, style, ...rest } = props;

  return (
    <Pressable style={[styles.button, style as StyleProp<ViewStyle>]} {...rest}>
      <Text style={[styles.buttonText, textStyle]}>{text}</Text>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  button: {
    alignItems: "center",
    padding: Indent.DEFAULT,
    justifyContent: "center",
    borderRadius: Radius.DEFAULT,
    backgroundColor: Color.BLUE_LIGHT
  },
  buttonText: {
    fontWeight: "500",
    color: Color.WHITE,
    fontSize: FontSize.DEFAULT
  }
});
