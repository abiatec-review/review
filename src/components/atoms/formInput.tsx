import React from "react";

import {
  StyleProp,
  StyleSheet,
  Text,
  TextInput,
  TextInputProps,
  View,
  ViewStyle
} from "react-native";

import { Color, FontSize, Indent, Radius } from "@utils";

interface Props {
  error?: string;
  inputStyle?: StyleProp<ViewStyle>;
}

export function FormInput(props: TextInputProps & Props) {
  const { error, style, inputStyle, ...rest } = props;

  const input = [styles.input, { borderColor: error ? Color.RED : Color.BLUE_LIGHT }, inputStyle];

  return (
    <View style={[styles.container, style]}>
      <TextInput {...rest} style={input} />
      {error && <Text style={styles.error}>{error}</Text>}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    borderRadius: Radius.DEFAULT
  },
  input: {
    borderWidth: 2,
    color: Color.BLACK,
    padding: Indent.DEFAULT,
    fontSize: FontSize.DEFAULT,
    borderRadius: Radius.DEFAULT
  },
  error: {
    color: Color.RED,
    textAlign: "right"
  }
});
