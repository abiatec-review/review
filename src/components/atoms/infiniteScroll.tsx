import React, { useEffect, useRef, useState } from "react";

import debounce from "lodash.debounce";
import { FlatList, FlatListProps, NativeScrollEvent, NativeSyntheticEvent } from "react-native";

import { Spinner } from "@components/atoms";
import { useOrientation } from "@hooks";
import { PagedData } from "@redux/models/entities";
import { Entity } from "@redux/models/entities";

interface Props<T extends Entity> {
  offset?: number;
  data: PagedData<T>;
  onScroll?: (offset: number) => void;
  load: (page: number) => Promise<void>;
  numColumns?: { portrait: number; landscape: number };
}

type DefaultProps<T extends Entity> = Omit<FlatListProps<T>, "numColumns" | "onScroll" | "data">;

export function InfiniteScroll<T extends Entity>(props: DefaultProps<T> & Props<T>) {
  const { load, numColumns, offset, onScroll, data, ...rest } = props;
  const { nextPage, hasMore, items } = data;

  const [isLoading, setIsLoading] = useState(false);

  const fetchData = () => {
    setIsLoading(true);
    debounce(() => {
      load(nextPage).then(() => setIsLoading(false));
    }, 2000)();
  };

  const onEndReached = () => {
    !isLoading && hasMore && fetchData();
  };

  useEffect(() => {
    fetchData();
  }, []);

  const { isPortrait } = useOrientation();

  const listRef = useRef<FlatList>(null);

  useEffect(() => {
    offset && listRef.current?.scrollToOffset({ offset });
  }, [isPortrait]);

  const scrollHandler = ({ nativeEvent }: NativeSyntheticEvent<NativeScrollEvent>) => {
    const ofset = nativeEvent.contentOffset.y;
    const { height, width } = nativeEvent.layoutMeasurement;
    const position = ofset * (width / height);
    onScroll && onScroll(position);
  };

  return (
    <FlatList
      data={items}
      ref={listRef}
      style={{ padding: 10 }}
      scrollEventThrottle={16}
      key={Number(isPortrait)}
      onScroll={scrollHandler}
      onEndReached={onEndReached}
      keyExtractor={(_, i) => i.toString()}
      contentContainerStyle={{ paddingBottom: 70 }}
      ListFooterComponent={isLoading ? <Spinner /> : null}
      numColumns={isPortrait ? numColumns?.portrait : numColumns?.landscape}
      {...rest}
    />
  );
}
