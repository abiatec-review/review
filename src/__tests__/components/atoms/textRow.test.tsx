import React from "react";

import { create } from "react-test-renderer";

import { TextRow } from "@components/atoms";

jest.mock("@react-native-firebase/auth", () => {});
jest.mock("@react-native-firebase/storage", () => {});

test("TextRow render", () => {
  const tree = create(<TextRow field="test" data="test" />).toJSON();
  expect(tree).toMatchSnapshot();
});
