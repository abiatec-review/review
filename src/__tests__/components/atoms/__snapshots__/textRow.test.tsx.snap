// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextRow render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 18,
        "marginRight": 20,
      }
    }
  >
    test:
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "flex": 1,
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
    test
  </Text>
</View>
`;
